src/archive.cpp

```diff
@@ -99,25 +99,17 @@ namespace zim
     throw EntryNotFound("No favicon found.");
   }
 
-  unsigned int parseIllustrationPathToSize(const std::string& str) {
-    constexpr auto illuSize = 13; // strlen("Illustration_");
-    auto xpos = str.find('x', illuSize);
-    if (xpos == std::string::npos) {
-      throw std::runtime_error("");
+  unsigned int parseIllustrationPathToSize(const std::string& s) {
+    int nw, nh, nEnd;
+    long int w, h;
+    if ( sscanf(s.c_str(), "Illustration_%n%ldx%n%ld@1%n)", &nw, &w, &nh, &h, &nEnd) == 2
+         && nEnd == s.size() && !isspace(s[nw]) && !isspace(s[nh]) && w == h && w >= 0) {
+        return (unsigned int)w;
     }
```

Comment:
Nice improvement!

I figured out one more potential issue - since %n doesn't count against conversions returned as a result of `scanf()`, inputs like `"Illustration_NNxNN"` followed by anything other than `"@1"` may also be randomly accepted (if the uninitialized value of `nEnd` happens to be equal to 2). The fix is easy - initialize `nEnd` to 0.

Reply:
Done